### PR TITLE
Fix a bunch of URL-reflecting IDL attributes

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/18825)

### CODE REVIEW

**Comment by nox** [comment]
This probably change some test expectations.

@bors-servo try

File: components/script/dom/macros.rs [hunk 1]
```diff
@@ -194,11 +194,8 @@ macro_rules! make_url_setter(
         fn $attr(&self, value: DOMString) {
             use dom::bindings::inheritance::Castable;
             use dom::element::Element;
-            use dom::node::document_from_node;
-            let value = AttrValue::from_url(document_from_node(self).url(),
```
Comment (emilio) [inline components/script/dom/macros.rs]: Eeek, this was so borked.

File: components/script/dom/htmlbaseelement.rs [hunk 2]
```diff
@@ -66,24 +66,24 @@ impl HTMLBaseElement {
...
-        let url = self.upcast::<Element>().get_url_attribute(&local_name!("href"));
-
-        // Step 4.
-        let url_record = fallback_base_url.join(&*url);
-
-        // Step 5, 6.
-        DOMString::from(url_record.as_ref().map(|url| url.as_str()).unwrap_or(""))
+        let url_record = document.fallback_base_url().join(url);
+
+        if let Ok(url_record) = url_record {
```
Comment (emilio) [inline components/script/dom/htmlbaseelement.rs]: nit: I'd just use match here instead if if let (..) else.

Comment (emilio) [reply components/script/dom/htmlbaseelement.rs]: That incidentally allows you to reorder the steps

**Review by emilio — APPROVED** [review]
Looks good... Tests? :-)

**Comment by emilio** [comment]
Oh, ok, let's wait for those test expectations to come back :P

**Comment by nox** [comment]
@bors-servo r=emilio

3 more tests now pass.

**Pull request merged**